scheduler/reconcile_util.go

```diff
@@ -225,8 +225,20 @@ func (a allocSet) filterByTainted(taintedNodes map[string]*structs.Node, support
 	ignore = make(map[string]*structs.Allocation)
 
 	for _, alloc := range a {
-		// Terminal allocs are always untainted as they should never be migrated.
-		if alloc.TerminalStatus() {
+		reconnected := false
+		expired := false
+
+		// Don't compute reconnected for these statuses since they cannot be reconnected.
+		if supportsDisconnectedClients &&
+			alloc.ClientStatus != structs.AllocClientStatusPending &&
+			alloc.ClientStatus != structs.AllocClientStatusComplete &&
+			alloc.ClientStatus != structs.AllocClientStatusLost {
+			reconnected, expired = alloc.Reconnected()
+		}
```

Comment:
We only want to potentially reconnect `running` and `unknown`, right? Maybe we should say that rather than what we don't want?